conda/resolve.py

```diff
@@ -647,16 +647,19 @@ def filter_group(_specs):
         # chance after their first "False" reduction. This catches more instances
         # where one package's filter affects another. But we don't have to be
         # perfect about this, so performance matters.
+        pruned_to_zero = set()
         for _ in range(2):
             snames.clear()
             slist = deque(explicit_specs)
-            reduced = False
             while slist:
                 s = slist.popleft()
-                top_level_spec = s
-                reduced = filter_group([s])
-                if reduced:
+                if filter_group([s]):
                     slist.append(s)
+                else:
+                    pruned_to_zero.add(s)
+
+        if pruned_to_zero and exit_on_conflict:
+            return {}
```

Comment:
@msarahan this is a good improvement! a couple questions:
a) what do you think about throwing an exception here, so that you could include the contents of `pruned_to_zero` in the error message? This gives users of unsatisfiable_hints = False a tidbit of information of which packages would be unsatisfiable.
b) why is this not the default behavior regardless of `exit_on_conflict`? it seems that if we've pruned an explicit_spec to 0, it's not likely to show up in the reduced index and the sat solver should fail, right?